File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/account/AccountPersistenceAdapter.java
package kyonggiyo.persistence.account;

import kyonggiyo.application.auth.port.outbound.LoadAccountPort;
import kyonggiyo.application.auth.port.outbound.SaveAccountPort;
import kyonggiyo.application.auth.domain.entity.Account;
import kyonggiyo.application.auth.domain.vo.Platform;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.Optional;

@Component
@RequiredArgsConstructor
public class AccountPersistenceAdapter implements LoadAccountPort, SaveAccountPort {

    private final AccountRepository repository;

    @Override
    public Optional<Account> findById(Long accountId) {
        return repository.findById(accountId);
    }

    @Override
    public Optional<Account> findByPlatformAndPlatformId(Platform platform, String platformId) {
        return repository.findByPlatformAndPlatformId(platform, platformId);
    }

    @Override
    public Account save(Account account) {
        return repository.save(account);
    }

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/account/AccountPersistenceAdapter.kt
package kyonggiyo.persistence.account

import kyonggiyo.application.auth.domain.entity.Account
import kyonggiyo.application.auth.domain.vo.Platform
import kyonggiyo.application.auth.port.outbound.LoadAccountPort
import kyonggiyo.application.auth.port.outbound.SaveAccountPort
import org.springframework.stereotype.Component

@Component
class AccountPersistenceAdapter(
        private val repository: AccountRepository
) : LoadAccountPort, SaveAccountPort {

    override fun findById(accountId: Long): Account? {
        return repository.findById(accountId)
    }

    override fun findByPlatformAndPlatformId(platform: Platform, platformId: String): Account? {
        return repository.findByPlatformAndPlatformId(platform, platformId)
    }

    override fun save(account: Account): Account {
        return repository.save(account)
    }

}
